src/aliceVision/mesh/Mesh.cpp

```diff
@@ -2529,9 +2535,82 @@ void Mesh::load(const std::string& filepath)
             nodes.push_back(node->mChildren[idChild]);
         }
     }
+
+    // merge coincident verts and update triangle ids
+    // keep uvs and normals as is to allow for face varying data
+    if (mergeCoincidentVerts)
+    {
+        std::map<int, int> oldToNewMap;
+        StaticVector<Point3d> uniquePoints;
+        for(int i = 0; i < pts.size(); ++i)
+        {
+            const Point3d& p = pts[i];
+            const auto it = std::find(uniquePoints.begin(), uniquePoints.end(), p);
+            if(it == uniquePoints.end())
+            {
+                oldToNewMap[i] = uniquePoints.size();
+                uniquePoints.push_back(p);
+            }
+            else
+            {
+                oldToNewMap[i] = static_cast<int>(std::distance(uniquePoints.begin(), it));
+            }
+        }
+
+        pts = uniquePoints;
+        for(triangle& f : tris)
+        {
+            f.v[0] = oldToNewMap[f.v[0]];
+            f.v[1] = oldToNewMap[f.v[1]];
+            f.v[2] = oldToNewMap[f.v[2]];
+        }
+    }
+
+    // set number of materials used
+    const std::unordered_set<int> materialIds = std::unordered_set<int>(_trisMtlIds.begin(), _trisMtlIds.end());
+    nmtls = static_cast<int>(materialIds.size());
```

Comment:
nmtls property was never set, so when loading an asset with materials the number of atlases wasn't being set correctly.